File: ABC/053/B.py
s = input()


def main(s):
    a_index = 200000 
    z_index = -1
    i = 0

    for c in s:
        if c == 'A':
            a_index = min(a_index, i)
        elif c == 'Z':
            z_index = max(z_index, i)
        i += 1

    return (z_index - a_index) + 1


print(main(s))

File: ABC/053/B.cs
using System;

class AtCoder
{
    static int Solve(string s)
    {
        int aIndex = -1;
        int zIndex = -1;
        int i = 0;

        foreach (char c in s)
        {
            if (c == 'A' && aIndex == -1)
            {
                aIndex = i;
            }
            else if (c == 'Z')
            {
                zIndex = i;
            }
            ++i;
        }

        return (zIndex - aIndex) + 1;
    }

    static void Main()
    {
        string s = Console.ReadLine();
        int answer = Solve(s);
        Console.WriteLine(answer);
    }
}
